stl/inc/xstring

```diff
@@ -110,8 +110,8 @@ struct _Char_traits { // properties of a string or stream element
                     _First1[_Idx] = _First2[_Idx];
                 }
             } else {
-                for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
-                    _First1[_Count - 1 - _Idx] = _First2[_Count - 1 - _Idx];
+                for (size_t _Idx = _Count; _Idx > 0; --_Idx) {
+                    _First1[_Idx - 1] = _First2[_Idx - 1];

```

Comment:
Does this actually provide superior codegen?

Reply:
This code is only executed at compile-time.

Reply:
I wrote this code in @miscco's #491 (in commit https://github.com/microsoft/STL/pull/491/commits/fd291a8991282cead9b160bb71e024758d7b9e71 ). Looking at it now, I can't remember why I wrote it as computing `[_Count - 1 - _Idx]`, and I agree that the proposed change is a clarity improvement (even if it doesn't affect runtime perf or compiletime throughput). There's no need for the "loop forward" and "loop backward" cases to have exactly the same `for` loop, there's no concern about `size_t` wrapping around (the new code cleanly stops when `_Idx` falls to `0`, and does nothing if it starts at `0`), and no danger of accessing `[-1]`. I think I simply missed how this could be clearly written with `[_Idx - 1]`.

I believe that we should merge this change (with the other one dropped, and the PR title appropriately changed). Thanks for finding and improving this weird bit of code that I wrote! :smile_cat:

Reply:
Fixed